Libs/Core/ctkJobScheduler.cpp

```diff
@@ -247,51 +232,71 @@ void ctkJobSchedulerPrivate::removeJobs(const QStringList &jobUIDs)
     foreach (QString jobUID, jobUIDs)
     {
       QSharedPointer<ctkAbstractJob> job = this->JobsQueue.value(jobUID);
-      if (!job || !this->JobsConnections.contains(jobUID))
+      if (!job)
       {
         continue;
       }
 
       datas.append(job->toVariant());
-
-      QMap<QString, QMetaObject::Connection> connections = this->JobsConnections.value(jobUID);
-      QObject::disconnect(connections.value("started"));
-      QObject::disconnect(connections.value("userStopped"));
-      QObject::disconnect(connections.value("finished"));
-      QObject::disconnect(connections.value("attemptFailed"));
-      QObject::disconnect(connections.value("failed"));
-      QObject::disconnect(connections.value("progress"));
-
-      this->JobsConnections.remove(jobUID);
-      this->JobsQueue.remove(jobUID);
+      job->freeUsedResources();
     }
   }
 
   emit q->jobUserStopped(datas);
 }
 
 //------------------------------------------------------------------------------
-void ctkJobSchedulerPrivate::removeAllJobs()
+bool ctkJobSchedulerPrivate::removeJob(const QString& jobUID)
 {
-  Q_Q(ctkJobScheduler);
+  logger.debug(QString("ctkJobScheduler: deleting job object %1 in thread %2.\n")
+    .arg(jobUID)
+    .arg(QString::number(reinterpret_cast<quint64>(QThread::currentThreadId()), 16)));
 
   {
     // The QMutexLocker is enclosed within brackets to restrict its scope and
     // prevent conflicts with other QMutexLockers within the scheduler's methods.
     QMutexLocker locker(&this->QueueMutex);
-    foreach (QSharedPointer<ctkAbstractJob> job, this->JobsQueue)
+    QSharedPointer<ctkAbstractJob> job = this->JobsQueue.value(jobUID);
+    if (!job || !this->JobsConnections.contains(jobUID))
     {
-      if (!job)
-      {
-        continue;
-      }
+      return false;
+    }
 
-      QString jobUID = job->jobUID();
-      if (!this->JobsConnections.contains(jobUID))
+    QMap<QString, QMetaObject::Connection> connections = this->JobsConnections.value(jobUID);
+    QObject::disconnect(connections.value("started"));
+    QObject::disconnect(connections.value("userStopped"));
+    QObject::disconnect(connections.value("finished"));
+    QObject::disconnect(connections.value("attemptFailed"));
+    QObject::disconnect(connections.value("failed"));
+    QObject::disconnect(connections.value("progress"));
+
+    this->JobsConnections.remove(jobUID);
+    this->JobsQueue.remove(jobUID);
+  }
+
+  this->onQueueJobsInThreadPool();
+  return true;
+}
+
+//------------------------------------------------------------------------------
+void ctkJobSchedulerPrivate::removeJobs(const QStringList &jobUIDs)
+{
+  QList<QVariant> datas;
```

Comment:
`data` is already plural, you can use `data` or you want to emphasize that there are many members then you can use something like `dataObjects`, `dataMembers`, ...